packages/image_picker/image_picker_android/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -716,22 +733,14 @@ private void handleChooseMultiImageResult(int resultCode, Intent intent) {
 
   private void handleChooseVideoResult(int resultCode, Intent data) {
     if (resultCode == Activity.RESULT_OK && data != null) {
-      Uri uri = data.getData();
-      // On several pre-Android 13 devices using Android Photo Picker, the Uri from getData() could be null.
-      if (uri == null) {
-        ClipData clipData = data.getClipData();
-        if (clipData != null && clipData.getItemCount() == 1) {
-          uri = clipData.getItemAt(0).getUri();
-        }
-      }
+      ArrayList<MediaPath> paths = getPathsFromIntent(data, false);
       // If there's no valid Uri, return an error
-      if (uri == null) {
+      if (paths == null || paths.size() < 1) {
         finishWithError("no_valid_video_uri", "Cannot find the selected video.");
         return;
       }
 
-      String path = fileUtils.getPathFromUri(activity, uri);
-      handleVideoResult(path);
+      handleVideoResult(paths.get(0).path);
```

Comment:
This part is the one piece that is a little awkward from forcing the path getting into one common method. It could also probably be fixed by making changes to `handleVideoResult`, though

Reply:
It should actually be trivial to avoid this, because I already standardized on returning lists for everything during the Pigeon migration. `handleVideoResult` can be removed, and you can just call `finishWithListSuccess` here, and change the one other use to `finishWithSuccess` (which is all `handleVideoResult` does now).

Reply:
`finishWithListSuccess` takes a list of strings, and we have a list of `MediaPath`s, so I just called `finishWithSuccess` directly in both places, and then removed `handleVideoResult`.